grr/server/grr_response_server/bin/grrafana.py

```diff
@@ -51,22 +73,134 @@ class JSONResponse(werkzeug_wrappers_json.JSONMixin,
 
   def __init__(self, response=None, *args, **kwargs) -> None:
     kwargs["mimetype"] = JSON_MIME_TYPE
-    if response is not None and not isinstance(response, werkzeug_wsgi.ClosingIterator):
+    if response is not None and not isinstance(response,
+                                               werkzeug_wsgi.ClosingIterator):
       response = json.dumps(response)
     super().__init__(response, *args, **kwargs)
 
 
+class Metric(abc.ABC):
+  """A single metric that can be fetched from a
+  Fleetspeak-based GRR deployment."""
+
+  @abc.abstractmethod
+  def __init__(self, name: str) -> None:
+    """Initializes a new metric."""
+    self._name = name
```

Comment:
Methods are either abstract or have some implementation. In general, constructors should not be abstract (since interfaces specify the that instances should fulfill but not how they should be constructed).